presto-hive/src/main/java/com/facebook/presto/hive/ConfigBasedCacheQuotaProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.cache.CacheConfig;
+import com.google.inject.Inject;
+import io.airlift.units.DataSize;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConfigBasedCacheQuotaProvider
+        implements CacheQuotaProvider
```

Comment:
Actually maybe we should name this `CacheQuotaRequirementProvider`? We already have a `CacheQuota` class, so maybe not make it more confusing.